pkg/quay/quay.go

```diff
@@ -96,6 +96,44 @@ func (c *QuayClient) CreateRepository(r RepositoryRequest) (*Repository, error)
 	return data, nil
 }
 
+// IsRepositoryExists checks if the specified image repository exists in quay.
+func (c *QuayClient) IsRepositoryExists(organization, imageRepository string) (bool, error) {
+	url := fmt.Sprintf("%s/repository/%s/%s", c.url, organization, imageRepository)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("%s %s", "Bearer", c.AuthToken))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("failed to Do request: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode == 404 {
+		return false, fmt.Errorf("Not Found")
```

Comment:
could we add more info to the error message here? (org, image repository)

Reply:
Kept it short intentionally so that when we use this routine in our e2e-tests, we will compare this returned error message with `Not Found` string to confirm when the repo does not exists, similar to [here](https://github.com/redhat-appstudio/image-controller/pull/24/files#diff-f49653252dd0f96d650015e2ebd690a20d9e7872dc1ebefc9c671da72d6a9f19R63). If we add additional info, that will make comparison complex. WDYT?

Reply:
my point was that when the error is hit, it's good when it's as informative as possible. 

> If we add additional info, that will make comparison complex. WDYT?

instead of checking the whole error, could we check its prefix instead, like `strings.HasPrefix()`, wdyt